feature-utils/poly-look/src/react-components/overlays/sideSlider.jsx

```diff
@@ -0,0 +1,422 @@
+  open = false,
+  onClose,
+  leftDistance = "124px",
+  backdropColor = [0, 0, 0, 0.3],
+  screenMaxWidth = "var(--max-width)",
+  animationDuration = "0.6s",
+  Component,
+  lastChildSelector,
+}) => {
+  let [red, green, blue, alpha] = backdropColor;
+  let commonStyles = {
+    duration: { transitionDuration: `${animationDuration}` },
+    noInteraction: { pointerEvents: "none" },
+    interaction: { pointerEvents: "auto" },
+    backdropOut: { backgroundColor: `rgba(${red}, ${green}, ${blue}, 0)` },
+    backdropIn: {
+      backgroundColor: `rgba(${red}, ${green}, ${blue}, ${alpha})`,
+    },
+    contentsIn: { transform: `translateX(${leftDistance})` },
+    contentsOutRight: { transform: `translateX(${screenMaxWidth})` },
+    backgroundTransition: { transition: "background-color" },
+    contentsTransition: { transition: "transform" },
+    noScroll: { overflowY: "hidden" },
+  };
+
+  const [step, updateStep] = useState(null);
+  const [isPressing, updateMousePress] = useState(false);
+  const [contentsStyle, updateContentsStyle] = useState(
+    commonStyles.contentsOutRight
+  );
+  const [backdropStyle, updateBackdropStyle] = useState({});
+  const [verticalGesture, updateVerticalGesture] = useState("scroll");
+  const [childScroll, updateChildScroll] = useState({});
+  const [movement, updateMovement] = useState({
+    startX: null,
+    startY: null,
+    currentX: null,
+    currentY: null,
+    swipeY: null,
+    direction: null,
+  });
+  const [axis, setAxis] = useState("");
+  const [animationsDoneCount, updateAnimationsStatus] = useState(0);
+  let contentsRef = useRef();
+
+  useEffect(() => {
+    let lastChild = contentsRef.current.querySelector(lastChildSelector);
+    if (lastChild) observe(lastChild);
+    else updateVerticalGesture("swipe");
+
+    width = window.innerWidth;
+    updateContentsStyle({
+      ...contentsStyle,
+      width: `calc(${width}px - ${leftDistance})`,
+    });
+
+    updateBackdropStyle(commonStyles.backdropOut);
+  }, []);
+
+  function observe(elem) {
+    observer = new IntersectionObserver(
+      ([entry]) => {
+        if (entry.isIntersecting) {
+          updateVerticalGesture("swipe");
+        } else {
+          updateVerticalGesture("scroll");
+        }
+      },
+      {
+        threshold: [1],
+      }
+    );
+    observer.observe(elem);
+  }
+
+  useEffect(() => {
+    if (open) {
+      updateStep(steps.in);
+    } else updateStep(steps.out);
+  }, [open]);
+
+  useEffect(() => {
+    if (step == steps.in) {
+      updateContentsStyle({
+        ...contentsStyle,
+        ...commonStyles.contentsIn,
+        ...commonStyles.contentsTransition,
+        ...commonStyles.duration,
+      });
+      updateBackdropStyle({
+        ...commonStyles.backdropIn,
+        ...commonStyles.backgroundTransition,
+        ...commonStyles.duration,
+        ...commonStyles.noInteraction,
+      });
+    }
+
+    if (step == steps.interactive) {
+      if (!stPoz) {
```

Comment:
So you're updating a global variable here. There's only one possible stPoz for all possible slides?

Reply:
That was a genuine mistake. But it's fixed now.